### ISSUE TITLE
no HELIUM information after update to .25

### ISSUE THREAD

**Issue by chapy23** [NebraLtd/helium-syncrobit#10]
![image](https://user-images.githubusercontent.com/11875678/227565774-b0aaaaf2-5e42-4a5a-9442-a88c276b15c9.png)

**Comment by chapy23** [comment]
I should have add that i use Balena Cloud to update.

**Comment by Dani1914** [comment]
Firmware Version | 2023.02.07.0-25 (c165b61)
With last update no more poc!!!
Edit-After reboot poc start.

**Comment by shawaj** [comment]
Please can you add gh_shawaj as a user on your BalenaCloud fleet @chapy23 ?

@Dani1914 how did you install? Using BalenaCloud?

**Comment by chapy23** [comment]
i'll do that tonight or tomorow morning.

bye

On March 24, 2023 7:36:02 p.m. EDT, Aaron Shaw ***@***.***> wrote:
>Please can you add gh_shawaj as a user on your BalenaCloud fleet @chapy23 ?
>
>@Dani1914 how did you install? Using BalenaCloud?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/NebraLtd/helium-syncrobit/issues/10#issuecomment-1483590431
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

-- 
Android with K-9 Mail, excuse my brevity.

**Comment by shawaj** [comment]
Ok let me know when done so I can take a look at the device and see what's happening

**Comment by chapy23** [comment]
i add you in my Balena account,  you can have a look now.

thank's

On March 24, 2023 7:57:07 p.m. EDT, Aaron Shaw ***@***.***> wrote:
>Ok let me know when done so I can take a look at the device and see what's happening 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/NebraLtd/helium-syncrobit/issues/10#issuecomment-1483605828
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

-- 
Android with K-9 Mail, excuse my brevity.

**Comment by chapy23** [comment]
I will glad to know how you troubleshoot my device. Do you use the shell access to the device ?

I love to learn new stuff,  thank's

bye

On March 24, 2023 7:57:07 p.m. EDT, Aaron Shaw ***@***.***> wrote:
>Ok let me know when done so I can take a look at the device and see what's happening 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/NebraLtd/helium-syncrobit/issues/10#issuecomment-1483605828
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>

-- 
Android with K-9 Mail, excuse my brevity.

**Comment by shawaj** [comment]
@chapy23 you added me as a member but you didn't give me access to any fleets.

Also, I need to have "developer" level access so I can change device settings to try and see what's going wrong

**Comment by chapy23** [comment]
hi there,

I think i made the change that will give you access you need.

bye
________________________________
De : Aaron Shaw ***@***.***>
Envoyé : 24 mars 2023 19:36
À : NebraLtd/helium-syncrobit ***@***.***>
Cc : chapy23 ***@***.***>; Mention ***@***.***>
Objet : Re: [NebraLtd/helium-syncrobit] no HELIUM information after update to .25 (Issue #10)


Please can you add gh_shawaj as a user on your BalenaCloud fleet @chapy23<https://github.com/chapy23> ?

@Dani1914<https://github.com/Dani1914> how did you install? Using BalenaCloud?

**Comment by MihaiDoro** [comment]
I have the same problem  as chapy23, no helium info after update to v25 with balenaCloud

**Comment by Maudslay** [comment]
I wish to add to this, no helium information either but guessing you guys already know that? any update on the fix patch? another note glad to see OTA is working :)

**Comment by shawaj** [comment]
@MihaiDoro @MihaiDoro @chapy23 this will hopefully be fixed with the next update once this PR is merged (hopefully later today)...
https://github.com/NebraLtd/hm-pyhelper/pull/234

I'm pretty sure it's working fine with this update after checking it on a test unit.

However, just FYI, this shouldn't cause an issue with mining just with the display of the correct info :-)

**Comment by shawaj** [comment]
The error message shows:

```
 diagnostics  INFO:root:Frequency: US915
 diagnostics  
 diagnostics  INFO:root:Diagnostics shipping not requested, skipping.
 diagnostics  INFO:root:Diagnostics complete
 diagnostics  INFO:root:Running periodic hardware diagnostics
 diagnostics  INFO:root:Running periodic hardware diagnostics
 diagnostics  2023-03-27 10:53:38,275 - [ERROR] - hm_pyhelper.lock_singleton - (lock_singleton.py).wrapper_lock_ecc -- /opt/python-dependencies/hm_pyhelper/lock_singleton.py:(67) - ECC is busy now.
 diagnostics  ERROR:hm_pyhelper.lock_singleton:ECC is busy now.
 diagnostics  2023-03-27 10:53:38,276 - [ERROR] - hm_pyhelper.lock_singleton - (lock_singleton.py).wrapper_lock_ecc -- /opt/python-dependencies/hm_pyhelper/lock_singleton.py:(67) - ECC is busy now.
 diagnostics  ERROR:hm_pyhelper.lock_singleton:ECC is busy now.
 diagnostics  2023-03-27 10:53:38,276 - [ERROR] - hw_diag.utilities.hardware - (hardware.py).get_public_keys_and_ignore_errors -- /opt/python-dependencies/hw_diag/utilities/hardware.py:(324) - 
 diagnostics  INFO:root:Running periodic hardware diagnostics
 diagnostics  2023-03-27 10:53:48,373 - [ERROR] - hm_pyhelper.lock_singleton - (lock_singleton.py).wrapper_lock_ecc -- /opt/python-dependencies/hm_pyhelper/lock_singleton.py:(67) - ECC is busy now.
 diagnostics  ERROR:hm_pyhelper.lock_singleton:ECC is busy now.
 diagnostics  2023-03-27 10:53:48,374 - [ERROR] - hm_pyhelper.lock_singleton - (lock_singleton.py).wrapper_lock_ecc -- /opt/python-dependencies/hm_pyhelper/lock_singleton.py:(67) - ECC is busy now.
 diagnostics  ERROR:hm_pyhelper.lock_singleton:ECC is busy now.
 diagnostics  2023-03-27 10:53:48,375 - [ERROR] - hw_diag.utilities.hardware - (hardware.py).get_public_keys_and_ignore_errors -- /opt/python-dependencies/hw_diag/utilities/hardware.py:(324) -
```

So it seems the ECC lock is interfering.

Should be solved by https://github.com/NebraLtd/hm-pyhelper/pull/234

**Comment by chapy23** [comment]
Thank's for your effort to fix that and also for the quick explanation.

I will wait for my miners to be updated OTA   🙂

bye
________________________________
De : Aaron Shaw ***@***.***>
Envoyé : 27 mars 2023 06:56
À : NebraLtd/helium-syncrobit ***@***.***>
Cc : chapy23 ***@***.***>; Mention ***@***.***>
Objet : Re: [NebraLtd/helium-syncrobit] no HELIUM information after update to .25 (Issue #10)


The error message shows:

 diagnostics  INFO:root:Frequency: US915
 diagnostics
 diagnostics  INFO:root:Diagnostics shipping not requested, skipping.
 diagnostics  INFO:root:Diagnostics complete
 diagnostics  INFO:root:Running periodic hardware diagnostics
 diagnostics  INFO:root:Running periodic hardware diagnostics
 diagnostics  2023-03-27 10:53:38,275 - [ERROR] - hm_pyhelper.lock_singleton - (lock_singleton.py).wrapper_lock_ecc -- /opt/python-dependencies/hm_pyhelper/lock_singleton.py:(67) - ECC is busy now.
 diagnostics  ERROR:hm_pyhelper.lock_singleton:ECC is busy now.
 diagnostics  2023-03-27 10:53:38,276 - [ERROR] - hm_pyhelper.lock_singleton - (lock_singleton.py).wrapper_lock_ecc -- /opt/python-dependencies/hm_pyhelper/lock_singleton.py:(67) - ECC is busy now.
 diagnostics  ERROR:hm_pyhelper.lock_singleton:ECC is busy now.
 diagnostics  2023-03-27 10:53:38,276 - [ERROR] - hw_diag.utilities.hardware - (hardware.py).get_public_keys_and_ignore_errors -- /opt/python-dependencies/hw_diag/utilities/hardware.py:(324) -
 diagnostics  INFO:root:Running periodic hardware diagnostics
 diagnostics  2023-03-27 10:53:48,373 - [ERROR] - hm_pyhelper.lock_singleton - (lock_singleton.py).wrapper_lock_ecc -- /opt/python-dependencies/hm_pyhelper/lock_singleton.py:(67) - ECC is busy now.
 diagnostics  ERROR:hm_pyhelper.lock_singleton:ECC is busy now.
 diagnostics  2023-03-27 10:53:48,374 - [ERROR] - hm_pyhelper.lock_singleton - (lock_singleton.py).wrapper_lock_ecc -- /opt/python-dependencies/hm_pyhelper/lock_singleton.py:(67) - ECC is busy now.
 diagnostics  ERROR:hm_pyhelper.lock_singleton:ECC is busy now.
 diagnostics  2023-03-27 10:53:48,375 - [ERROR] - hw_diag.utilities.hardware - (hardware.py).get_public_keys_and_ignore_errors -- /opt/python-dependencies/hw_diag/utilities/hardware.py:(324) -


So it seems the ECC lock is interfering.

Should be solved by NebraLtd/hm-pyhelper#234<https://github.com/NebraLtd/hm-pyhelper/pull/234>

**Comment by shawaj** [comment]
Great. I'll keep you updated here 😊

**Comment by vstrong-info** [comment]
> Thank's for your effort to fix that and also for the quick explanation. I will wait for my miners to be updated OTA 🙂 bye
> […](#)

I thought BalenaCloud deployed miners do not get OTA... :)

**Comment by shawaj** [comment]
@vstrong-info you are correct they don't unless you set it up yourself ☺️

**Comment by shawaj** [comment]
@chapy23 can you push the latest release from this repo to the fleet and see if it's fixed?

I don't seem to have access anymore

**Comment by chapy23** [comment]
yes, sorry i remove the access

i can do that, no problem

what is the best practice to push the new release
i want to learn what is the better way to do it, maybe there is just one  🙂

bye

________________________________
De : Aaron Shaw ***@***.***>
Envoyé : 28 mars 2023 11:37
À : NebraLtd/helium-syncrobit ***@***.***>
Cc : chapy23 ***@***.***>; Mention ***@***.***>
Objet : Re: [NebraLtd/helium-syncrobit] no HELIUM information after update to .25 (Issue #10)


@chapy23<https://github.com/chapy23> can you push the latest release from this repo to the fleet and see if it's fixed?

I don't seem to have access anymore

**Comment by shawaj** [comment]
@chapy23 easiest way is to go to https://github.com/NebraLtd/helium-syncrobit and click "deploy with Balena"

![Screenshot_20230328_172013_Chrome](https://user-images.githubusercontent.com/3359418/228304987-d4a7e63a-6c46-4ddd-93d7-39b3bb4962e8.jpg)

Then click to use an existing fleet and choose the correct one...

![Screenshot_20230324_132456_Chrome](https://user-images.githubusercontent.com/3359418/228305142-b3a5f950-a553-4595-9be5-95e4b2a06941.jpg)

Then toggle the advanced tab and make sure it's got all the info loaded...

![Screenshot_20230324_132522_Chrome](https://user-images.githubusercontent.com/3359418/228305281-a896d9a7-d413-4063-890d-640d34cb9e13.jpg)

**Comment by vstrong-info** [comment]
@chapy23 Follow the above instructions from @shawaj, push the new release to an existing Fleet and all your fleet devices will be updated. They will retain the names and passwords as well.

**Comment by vstrong-info** [comment]
> ![image](https://user-images.githubusercontent.com/11875678/227565774-b0aaaaf2-5e42-4a5a-9442-a88c276b15c9.png)

Version 26 fixes this issue.

**Comment by shawaj** [comment]
@vstrong-info but it's still showing "not available"?

Is that an old screenshot?

**Comment by chapy23** [comment]
Forget about delete screenshot, i was able to remove it by myself, sorry

Now, running:  2023.02.07.0-26 (7784b16)   update with "Deploy with Balena",  all good !
password and devicename still the same (not lost)
also confirm that Helium info are there.
anything i should look for you here  ?

**Comment by vstrong-info** [comment]
> 

Yes, it is @chapy23's screenshot from the first post.

**Comment by shawaj** [comment]
Ah ok, perfect. Looks like everything fixed now then yes. 🎉🥳🙌

Perfect. Will close this issue but if you spot any other problems feel free to open a new issue 👍
